plumbing/object/commit_walker_file.go

```diff
@@ -128,7 +128,9 @@ func isParentHash(hash plumbing.Hash, commit *Commit) bool {
 func (c *commitFileIter) ForEach(cb func(*Commit) error) error {
 	for {
 		commit, nextErr := c.Next()
-		if nextErr != nil {
+		if nextErr == io.EOF {
+			break
+		} else if nextErr != nil {
```

Comment:
remove this unnecessary else

Reply:
Thank you for the review. I've fixed it.